src/Cache/Redis.php

```diff
@@ -0,0 +1,161 @@
+<?php
+
+namespace Kirby\Cache;
+
+use Redis as RedisClient;
+use Throwable;
+
+/**
+ * Redis Cache Driver
+ *
+ * @package   Kirby Cache
+ * @author    Ahmet Bora <ahmet@getkirby.com>
+ * @link      https://getkirby.com
+ * @copyright Bastian Allgeier
+ * @license   https://opensource.org/licenses/MIT
+ */
+class Redis extends Cache
+{
+	/**
+	 * Store for the redis connection
+	 */
+	protected RedisClient $connection;
+
+	/**
+	 * Stores whether the connection was successful
+	 */
+	protected bool $enabled;
+
+	/**
+	 * Sets all parameters which are needed to connect to Redis
+	 *
+	 * @param array $options 'host'   (default: 127.0.0.1)
+	 *                       'port'   (default: 6379)
+	 */
+	public function __construct(array $options = [])
+	{
+		$options = [
+			'host' => '127.0.0.1',
+			'port' => 6379,
+			...$options
+		];
+
+		parent::__construct($options);
+
+		$allowed = [
+			'host',
+			'port',
+			'readTimeout',
+			'connectTimeout',
+			'persistent',
+			'auth',
+			'ssl',
+			'retryInterval',
+			'backoff'
+		];
+
+		// filter only redis supported keys
+		$redisOptions = array_intersect_key($options, array_flip($allowed));
+
+		// create redis connection
+		$this->connection = new RedisClient($redisOptions);
+
+		// set the prefix if defined
+		if ($prefix = $options['prefix'] ?? null) {
+			$this->connection->setOption(RedisClient::OPT_PREFIX, $prefix);
+		}
+
+		// select the database if defined
+		$database = $options['database'] ?? null;
+		if ($database !== null) {
```

Comment:
```suggestion
		if ($database = $options['database'] ?? null) {
```

Reply:
I was expecting this suggestion :) But it will not work since database can be integer. So if database index number is `0`, if condition won't work. `if (0) { }`

Reply:
Ah I see, no all good then as is